packages/blade/src/utils/isPartialMatchObjectKeys/isPartialMatchObjectKeys.ts

```diff
@@ -0,0 +1,76 @@
+/* eslint-disable no-console */
+import isObject from 'lodash/isObject';
+
+export type DeepPartial<T> = {
+  [P in keyof T]?: T[P] extends Record<number | string, unknown> ? DeepPartial<T[P]> : T[P];
+};
+
+const isPartialMatchObjectKeys = <ActualObject>({
+  objectToMatch,
+  objectToInspect,
+}: {
+  objectToMatch: DeepPartial<ActualObject>;
+  objectToInspect: ActualObject;
+}): boolean => {
+  const matchResponses: boolean[] = [];
+
+  const matchObjectKeys = ({
+    innerObjectToMatch,
+    innerObjectToInspect,
+  }: {
+    innerObjectToMatch: DeepPartial<ActualObject>;
+    innerObjectToInspect: DeepPartial<ActualObject>;
+  }): void => {
+    for (const [key, valueToMatch] of Object.entries(innerObjectToMatch)) {
+      const valueToInspect = innerObjectToInspect[key as keyof DeepPartial<ActualObject>];
+
+      if (innerObjectToInspect.hasOwnProperty(key)) {
+        if (
+          valueToMatch === null ||
+          valueToMatch === undefined ||
+          valueToMatch === '' ||
+          Array.isArray(valueToMatch) ||
+          // the condition checks if the "valueToMatch" is not of type object then "valueToMatch" type should be same as type of "valueToInspect"
+          (!(valueToMatch instanceof Object) && typeof valueToMatch !== typeof valueToInspect)
```

Comment:
The `typeof` match check can be done regardless, no?

Reply:
elaborate?